wrappers/python/tests/layer/test_batch_norm.py

```diff
@@ -0,0 +1,97 @@
+# @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+#                              (Skoltech), Russia. All rights reserved.
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/tests/layer/test_layer_norm.py
+# Test for nntile.layer.LayerNorm
+#
+# @version 1.0.0
+
+import nntile
+from nntile.layer import BatchNorm2d
+
+import nntile.utils
+import nntile.utils.constructors
+import numpy as np
+import torch
+import torch.nn as nn
+from dataclasses import dataclass
+
+config = nntile.starpu.Config(1, 0, 0)
+
+nntile.starpu.init()
+dtypes = [np.float32, np.float64]
+
+@dataclass
+class BatchNormTestParams:
+    shape: list
+    dtype: np.dtype
+
+    atol: float = 1e-05
+
+    eps: float = 1e-05
+    redux: bool = False
+
+def generate_input(params: BatchNormTestParams):
+    input_np = np.random.rand(*params.shape).astype(params.dtype)
+    output_grad_np = np.random.rand(*params.shape).astype(params.dtype)
+
+    input_torch = torch.tensor(input_np, requires_grad=True)
+    output_grad_torch = torch.tensor(output_grad_np, requires_grad=False)
+
+    # input_np = np.asfortranarray(input_np)
+    # output_grad_np = np.asfortranarray(output_grad_np)
+
+    input_nnt = nntile.tensor.tensor_from_numpy(input_np)
+    input_grad_nnt = nntile.tensor.tensor_from_numpy(np.zeros(params.shape).astype(params.dtype))
+    input_moment = nntile.tensor.TensorMoments(input_nnt, input_grad_nnt, grad_required=True)
+    
+    output_grad_nnt = nntile.tensor.tensor_from_numpy(output_grad_np)
+
+    return (input_moment, output_grad_nnt), (input_torch, output_grad_torch)
+
+
+def test_batchnorm(params: BatchNormTestParams):
+    (input_moment, output_grad_nnt), (input_torch, output_grad_torch) = generate_input(params)
+    
+    # torch forward/backward
+    o = nn.BatchNorm2d(params.shape[1], dtype=input_torch.dtype, affine=False)(input_torch)
```

Comment:
Extremely bad variable naming. It is okay use one letter names in short expression like list comprehension or loops.

Reply:
done